tools: add DM query status doc (#846)
pingcap/docs: tools/dm-query-status.md

                                                                                            # synchronize data.
                        "blockingDDLs": [       # The DDL list that is blocked currently. It is not empty only when all the upstream tables of this 
                                                # DM-worker are "synced". In this case, it indicates the sharding DDL statement to be executed or skipped.
                            "USE `test`; ALTER TABLE `test`.`t_target` DROP COLUMN `age`;"
                        ],
                        "unresolvedGroups": [   # The sharding group that is not resolved.
                            {
                                "target": "`test`.`t_target`",                  # The downstream database table to be synchronized.
                                "DDLs": [
                                    "USE `test`; ALTER TABLE `test`.`t_target` DROP COLUMN `age`;"
                                ],
                                "firstPos": "(bin|000001.000001, 3130)",        # The starting position of the sharding DDL statement.

>                                 "synced": [                                     # The sharded table of the sharding DDL statement that `Sync` has read.

Reviewer: The shared table that `sync` unit has read its sharding DDL statement.
The original Chinese is not accuracy, `the shared table` does not belong to `sharding DDL`.

Reviewer 2: The suggested version is still not appropriate in grammar. Please confirm the sentence logic with @amyangfei and provide a new version. :)

                                    "`test`.`t2`"
                                    "`test`.`t3`"
                                    "`test`.`t1`"
                                ],

>                                 "unsynced": [                                   # The upstream table of the sharding DDL statement that has not been

Reviewer: The upstream table that has not executed this sharding DDL

Reviewer 2: ```suggestion
                                "unsynced": [                                   # The upstream table that has not executed this sharding DDL statement.
```

>                                                                                 # executed. If any upstream tables have not been synchronized completely,  

Reviewer 2: ```suggestion
                                                                                # executed. If any upstream tables have not finished synchronization,  
```

                                                                                # `blockingDDLs` is empty.
                                ]
                            }
                        ],

>                         "synced": false         # Whether the progress of incremental synchronization in the downstream has been the same as that in the 

Reviewer: In fact there is no `incremental synchronization` in upstream. what about ``Whether the incremental synchronization catches up with upstream and has the same binlog position as that in the upstream``

                "relayBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-14",     # The GTID information of the binlog that has been pulled to the local 
                                                                                    # storage.
                "relayCatchUpMaster": true,     # Whether the progress of synchronizing the relay log in the local storage has been the same as that in 
                                                # the upstream.
                "stage": "Running",             # The status of the `Sync` processing unit of the relay log.
                "result": null
            }
        },
        {
            "result": true,
            "worker": "172.17.0.3:10081",
            "msg": "",
            "subTaskStatus": [
                {
                    "name": "test",
                    "stage": "Running",
                    "unit": "Load",
                    "result": null,
                    "unresolvedDDLLockID": "",
                    "load": {                   # The synchronization information of the `Load` processing unit.
                        "finishedBytes": "115", # The number of bytes that have been imported fully.

>                         "totalBytes": "452",    # The total number of bytes that need to be imported.

Reviewer: ```suggestion
                        "totalBytes": "452",    # The total number of bytes that need to be loaded.
```

>                         "progress": "25.44 %"   # The progress of the full import.

Reviewer: ```suggestion
                        "progress": "25.44 %"   # The progress of the load process.
```

Author: @amyangfei load process -> loading process

                    }
                }
            ],
            "relayStatus": {
                "masterBinlog": "(bin.000001, 28507)",
                "masterBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-96",
                "relaySubDir": "c0149e17-dff1-11e8-b6a8-0242ac110004.000001",
                "relayBinlog": "(bin.000001, 28507)",
                "relayBinlogGtid": "c0149e17-dff1-11e8-b6a8-0242ac110004:1-96",
                "relayCatchUpMaster": true,
                "stage": "Running",
                "result": null
            }
        },
        {
            "result": true,
            "worker": "172.17.0.3:10081",
            "msg": "",
            "subTaskStatus": [
                {
                    "name": "test",
                    "stage": "Running",
                    "unit": "Load",
                    "result": null,
                    "unresolvedDDLLockID": "",
                    "load": {                   # The synchronization information of the `Load` processing unit.

>                         "finishedBytes": "115", # The number of bytes that have been imported fully.

Reviewer: ```suggestion
                        "finishedBytes": "115", # The number of bytes that have been loaded.
```

>                         "totalBytes": "452",    # The total number of bytes that need to be imported.

Reviewer: ```suggestion
                        "totalBytes": "452",    # The total number of bytes that need to be loaded.
```

>                         "progress": "25.44 %"   # The progress of the full import.

Reviewer: ```suggestion
                        "progress": "25.44 %"   # The progress of the load process.
```

                            }
                        ],
                        "detail": null
                    },
                    "unresolvedDDLLockID": "",
                    "load": {
                        "finishedBytes": "0",
                        "totalBytes": "156",
                        "progress": "0.00 %"
                    }
                }
            ],
            "relayStatus": {
                "masterBinlog": "(bin.000001, 1691)",
                "masterBinlogGtid": "97b5142f-e19c-11e8-808c-0242ac110005:1-9",
                "relaySubDir": "97b5142f-e19c-11e8-808c-0242ac110005.000001",
                "relayBinlog": "(bin.000001, 1691)",
                "relayBinlogGtid": "97b5142f-e19c-11e8-808c-0242ac110005:1-9",
                "relayCatchUpMaster": true,
                "stage": "Running",
                "result": null
            }
        }
    ]
}

```

For the status description and status switch relationship of "stage" of "subTaskStatus" of "workers", see [Subtask status](#subtask-status).

> For operation details of "unresolvedDDLLockID" of "subTaskStatus" of "workers", see [Troubleshooting Sharding DDL Locks](../tools/troubleshooting-sharding-ddl-locks/).

Reviewer: Where do LINE 184-185 come from, I can't find them in CN docs.
https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/task-handling/query-status.md

Author: @amyangfei They are in the above code block. Relative URLs are not user-friendly in the code block, so I move them out of this block and place them here.
